File: julia/monte_carlo_pi.jl
function monte_carlo_pi(n::Int)
    count = 0
    for _ in 1:n
        x = rand()
        y = rand()
        if x * x + y * y <= 1.0
            count += 1
        end
    end
    return 4.0 * count / n
end

n = 10_000_000
pi_est = monte_carlo_pi(n)
println("Pi ≈ $pi_est")

File: rust/src/monte_carlo_pi.rs
use rand::Rng;

fn monte_carlo_pi(n: i64) -> f64 {
    let mut rng = rand::thread_rng();
    let mut count: i64 = 0;
    for _ in 0..n {
        let x: f64 = rng.gen();
        let y: f64 = rng.gen();
        if x * x + y * y <= 1.0 {
            count += 1;
        }
    }
    4.0 * count as f64 / n as f64
}

fn main() {
    let n = 10_000_000;
    let pi_est = monte_carlo_pi(n);
    println!("Pi ≈ {:.10}", pi_est);
}
